cli/src/commands/diffedit.rs

```diff
@@ -136,7 +145,7 @@ don't make any changes, then the operation will be aborted.",
     };
     let base_tree = merge_commit_trees(tx.repo(), base_commits.as_slice())?;
     let tree = target_commit.tree()?;
-    let tree_id = diff_editor.edit(&base_tree, &tree, &EverythingMatcher, format_instructions)?;
+    let tree_id = diff_editor.edit(&base_tree, &tree, &matcher, format_instructions)?;
```

Comment:
[This FIXME](https://github.com/jj-vcs/jj/blob/4201af2107a8003e04ba81b68697eb6c577a5542/cli/src/merge_tools/mod.rs#L286C8-L292) makes me think that this doesn't do what you (and I) would want when using the internal diff editor. Can you check?

Reply:
I think we should fix it by making the internal tool match external tools, i.e. by making `edit_diff_builtin()` keep the right tree for unmatched paths. That could perhaps be done by making `apply_diff_builtin()` create the `tree_builder` from `right_tree` and adjusting `apply_changes()` accordingly. I'll let you figure out the details.